typescript/lib/callback.ts

```diff
@@ -0,0 +1,97 @@
+import { createLogger, format, transports } from "winston";
+
+// Constants
+const DEFAULT_TIMEOUT: number = 5; // Default timeout for callback execution in seconds
+
+export class CallbackEvent {
+  name: string;
+  file: string;
+  data: any;
+  ts_ns: number;
+
+  constructor(
+    name: string,
+    file: string,
+    data: any,
+    ts_ns: number = Date.now()
+  ) {
+    this.name = name;
+    this.file = file;
+    this.data = data;
+    this.ts_ns = ts_ns;
+  }
+}
+
+// Type Aliases
+type Callback = (event: CallbackEvent) => Promise<any>;
+// type Result = Ok<any> | Err<any>;
+
+async function executeCoroutineWithTimeout(
+  coroutine: Promise<any>,
+  timeout: number
+): Promise<any> {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(new Error("Timeout"));
+    }, timeout * 1000);
+
+    coroutine
+      .then((result) => {
+        resolve(result);
+      })
+      .catch((error) => {
+        resolve(error);
+      });
+  });
+}
+
+export class CallbackManager {
+  callbacks: Callback[];
+  results: any[];
+  timeout: number;
+
+  constructor(callbacks: Callback[], timeout: number = DEFAULT_TIMEOUT) {
+    this.callbacks = callbacks;
+    this.results = [];
+    this.timeout = timeout;
+  }
+
+  async runCallbacks(event: CallbackEvent): Promise<void> {
+    const tasks = this.callbacks.map((callback) =>
+      executeCoroutineWithTimeout(callback(event), this.timeout)
+    );
+    this.results = await Promise.all(tasks);
```

Comment:
If we rejected in `withTimeout` and still want to swallow errors, we could do something like:
```
// Use .allSettled if we want to avoid propagating errors and just log something
const results = await Promise.allSettled(promises);
results.forEach(result => {
  if (result.status === "rejected") {
    console.error('Error running callback for event: ', result.event, result.error);
   }
});
```

But fwiw, I think we should keep as Promise.all and just update `withTimeout` to reject so that it throws the error instead of swallowing it

Reply:
Updated withTimeout as advised, throws reject instead of swallowing it